src/main/java/frc/robot/subsystems/ShooterHood.java

```diff
@@ -0,0 +1,64 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2019 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.subsystems;
+
+import edu.wpi.first.wpilibj.Solenoid;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.Constants.RobotType;
+
+public class ShooterHood extends SubsystemBase {
+
+  private static final int liftSolenoidChannel = 6;
+  private static final int stopSolenoidChannel = 5;
+
+  private Solenoid liftSolenoid;
+  private Solenoid stopSolenoid;
+
+  /**
+   * Creates a new ShooterHood.
+   */
+  public ShooterHood() {
+    if (!available()) {
+      return;
+    }
+    liftSolenoid = new Solenoid(liftSolenoidChannel);
+    stopSolenoid = new Solenoid(stopSolenoidChannel);
+  }
+
+  private boolean available() {
+    return Constants.getRobot() == RobotType.ROBOT_2020;
+  }
+
+  @Override
+  public void periodic() {
+    // This method will be called once per scheduler run
+  }
+
+  /**
+   * Sets the position of the lift solenoid
+   * 
+   * @param raised
+   */
+  public void setLiftPosition(boolean raised) {
+    if (available()) {
+      liftSolenoid.set(raised);
+    }
+  }
+
+  /**
+   * Sets whether the stop solenoid is extended
+   * 
+   * @param stopped
+   */
+  public void setStopPosition(boolean stopped) {
+    if (available()) {
+      stopSolenoid.set(stopped);
+    }
+  }
```

Comment:
This should probably check the state of the lift piston to prevent extending the stop if the lift is lowered.

Reply:
If the lift is lowered? Isn't that what happens when moving to the middle position? It might make sense to prevent extending the stop if the lift is currently extended.

Reply:
Isn't that what normally happens to move to the middle position? I don't really understand the reasoning behind this.

Reply:
I meant if the lift is raised.